File: services/typescript/koa/src/routes/middleware/invalid_path.ts
import type { Middleware } from "koa";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "telemetry";

export const invalidPathMiddleware: Middleware = (ctx) => {
  getLogger().warn({ method: ctx.req.method, path: ctx.path }, "Invalid path");

  sendError(ctx, AppErrors.InvalidPath);
};

File: services/python/quart/src/routes/handlers/invalid_path.py
import structlog
from quart import request
from werkzeug.exceptions import HTTPException

from routes.errors import AppErrors, send_error


async def invalid_path_handler(_error: HTTPException):
    structlog.get_logger().warn(
        "Invalid path", method=request.method, path=request.path
    )
    return send_error(AppErrors.InvalidPath)
